### PR TITLE
Adds Resource Trigger Hotkey to HUD/TUI

### PR DESCRIPTION
**Pull request by bshore**
## Notes

Aims to resolve https://github.com/tilt-dev/tilt/issues/2995 by introducing hotkey `t` to trigger rebuild on currently selected resource when in HUD / TUI mode.

image

### CODE REVIEW

File: internal/hud/hud.go [hunk 1]
```diff
@@ -174,6 +178,22 @@ func (h *Hud) handleScreenEvent(ctx context.Context, dispatch func(action store.
 				escape()
 			case r == 'R': // hidden key for recovering from printf junk during demos
 				h.r.screen.Sync()
+			case r == 't': // [T]rigger resource update
+				_, selected := h.selectedResource()
+				h.recordInteraction("trigger_resource")
+				payload := []byte(fmt.Sprintf(`{"manifest_names":[%q], "build_reason": %d}`, selected.Name, model.BuildReasonFlagTriggerCLI))
+				resp, err := http.Post(fmt.Sprintf("http://%s/api/%s", h.webURL.Host, "trigger"), "application/json", bytes.NewBuffer(payload))
```
Comment (bshore) [inline internal/hud/hud.go:185]: Note: I tried to use `dispatch()` passing this Action:

https://github.com/tilt-dev/tilt/blob/688f5dfb982ce68de84977dc26e3d12b7f486355/internal/hud/server/actions.go#L7-L10

Like how the server handles it:

https://github.com/tilt-dev/tilt/blob/688f5dfb982ce68de84977dc26e3d12b7f486355/internal/hud/server/server.go#L270

But I encountered `import cycle not allowed`, so I ended up following the example in the linked issue here - https://github.com/tilt-dev/tilt/issues/2894#issuecomment-585866835 to just send a POST

File: internal/hud/hud.go [hunk 2]
```diff
@@ -174,6 +178,22 @@ func (h *Hud) handleScreenEvent(ctx context.Context, dispatch func(action store.
 				escape()
 			case r == 'R': // hidden key for recovering from printf junk during demos
 				h.r.screen.Sync()
+			case r == 't': // [T]rigger resource update
+				_, selected := h.selectedResource()
+				h.recordInteraction("trigger_resource")
+				payload := []byte(fmt.Sprintf(`{"manifest_names":[%q], "build_reason": %d}`, selected.Name, model.BuildReasonFlagTriggerHUD))
+				resp, err := http.Post(fmt.Sprintf("http://%s/api/trigger", h.webURL.Host), "application/json", bytes.NewBuffer(payload))
```
Comment (bshore) [inline internal/hud/hud.go:185]: Note: I tried to use `dispatch()` passing this Action:

https://github.com/tilt-dev/tilt/blob/688f5dfb982ce68de84977dc26e3d12b7f486355/internal/hud/server/actions.go#L7-L10

Like how the server handles it:

https://github.com/tilt-dev/tilt/blob/688f5dfb982ce68de84977dc26e3d12b7f486355/internal/hud/server/server.go#L270

But I encountered `import cycle not allowed`, so I ended up following the example in the linked issue here - https://github.com/tilt-dev/tilt/issues/2894#issuecomment-585866835 to just send a POST

Comment (nicks) [reply internal/hud/hud.go:185]: how hard would it be to move AppendToTriggerQueueAction to the `internal/store` package? then you could use it here. POST is more of the api for external binaries, and using it here makes this harder to test/maintain.

**Review by nicks — APPROVED** [review]
LGTM!

**Pull request merged**